[MEDIUM] Clean up Python linting warnings - unused imports

## Description

The backend has 198 Python linting warnings (primarily unused imports) that should be cleaned up:

- Most are fixable with resume-api/api/async_pdf_routes.py:22:1: E402 Module level import not at top of file
   |
20 | sys.path.insert(0, str(lib_path))
21 |
22 | from lib.cli import ResumeGenerator
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
23 | from lib.queue import Job, JobState, JobPriority
24 | from config.dependencies import AuthorizedAPIKey, rate_limit
   |

resume-api/api/async_pdf_routes.py:23:1: E402 Module level import not at top of file
   |
22 | from lib.cli import ResumeGenerator
23 | from lib.queue import Job, JobState, JobPriority
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
24 | from config.dependencies import AuthorizedAPIKey, rate_limit
25 | from config import settings
   |

resume-api/api/async_pdf_routes.py:24:1: E402 Module level import not at top of file
   |
22 | from lib.cli import ResumeGenerator
23 | from lib.queue import Job, JobState, JobPriority
24 | from config.dependencies import AuthorizedAPIKey, rate_limit
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
25 | from config import settings
26 | from monitoring import logging_config
   |

resume-api/api/async_pdf_routes.py:25:1: E402 Module level import not at top of file
   |
23 | from lib.queue import Job, JobState, JobPriority
24 | from config.dependencies import AuthorizedAPIKey, rate_limit
25 | from config import settings
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
26 | from monitoring import logging_config
   |

resume-api/api/async_pdf_routes.py:26:1: E402 Module level import not at top of file
   |
24 | from config.dependencies import AuthorizedAPIKey, rate_limit
25 | from config import settings
26 | from monitoring import logging_config
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
27 |
28 | from .models import (
   |

resume-api/api/async_pdf_routes.py:28:1: E402 Module level import not at top of file
   |
26 |   from monitoring import logging_config
27 |
28 | / from .models import (
29 | |     SubmitPDFRenderJobRequest,
30 | |     SubmitPDFRenderJobResponse,
31 | |     JobStatusResponse,
32 | |     JobStatus,
33 | |     QueueStatsResponse,
34 | |     ErrorResponse,
35 | |     JobPriorityLevel,
36 | | )
   | |_^ E402
37 |
38 |   logger = logging_config.get_logger(__name__)
   |

resume-api/config/dependencies.py:100:16: F823 Local variable `verify_api_key` referenced before assignment
    |
 98 |     if settings.master_api_key:
 99 |         if is_hashed_key(settings.master_api_key):
100 |             if verify_api_key(x_api_key, settings.master_api_key):
    |                ^^^^^^^^^^^^^^ F823
101 |                 return x_api_key
102 |         else:
    |

resume-api/lib/utils/cache.py:588:1: E402 Module level import not at top of file
    |
588 | from functools import wraps
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
589 | from fastapi import Request, Response
    |

resume-api/lib/utils/cache.py:589:1: E402 Module level import not at top of file
    |
588 | from functools import wraps
589 | from fastapi import Request, Response
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
    |

resume-api/lib/utils/cache.py:589:21: F811 Redefinition of unused `Request` from line 18
    |
588 | from functools import wraps
589 | from fastapi import Request, Response
    |                     ^^^^^^^ F811
    |
    = help: Remove definition: `Request`

resume-api/lib/utils/cache.py:589:30: F811 Redefinition of unused `Response` from line 19
    |
588 | from functools import wraps
589 | from fastapi import Request, Response
    |                              ^^^^^^^^ F811
    |
    = help: Remove definition: `Response`

resume-api/tests/integration/test_api_key_management_e2e.py:324:1: E402 Module level import not at top of file
    |
323 | # Import User for test
324 | from database import User
    | ^^^^^^^^^^^^^^^^^^^^^^^^^ E402
    |

resume-api/tests/test_async_rendering.py:13:1: E402 Module level import not at top of file
   |
11 | sys.path.insert(0, str(resume_api_path))
12 |
13 | from main import app
   | ^^^^^^^^^^^^^^^^^^^^ E402
14 | from config.dependencies import get_api_key
   |

resume-api/tests/test_async_rendering.py:14:1: E402 Module level import not at top of file
   |
13 | from main import app
14 | from config.dependencies import get_api_key
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
15 |
16 | # Sample resume data for testing
   |

resume-api/tests/test_github_routes_v2.py:24:1: E402 Module level import not at top of file
   |
23 | # We need to mock the dependencies for testing
24 | from fastapi.security import HTTPBearer
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
25 |
26 | security = HTTPBearer()
   |

resume-cli/cli/utils/versioning.py:48:13: F841 Local variable `content` is assigned to but never used
   |
46 |         # Read current resume content
47 |         with open(self.yaml_path, "r", encoding="utf-8") as f:
48 |             content = f.read()
   |             ^^^^^^^ F841
49 |
50 |         # Get version number
   |
   = help: Remove assignment to unused variable `content`

resume_ai_lib/src/resume_ai_lib/ats_checker.py:460:13: F402 Import `field` from line 10 shadowed by loop variable
    |
459 |         basics = resume_data.get("basics", {})
460 |         for field in ["name", "summary", "headline"]:
    |             ^^^^^ F402
461 |             if basics.get(field):
462 |                 text_parts.append(str(basics[field]))
    |

scripts/load_test_replicas.py:317:13: F841 Local variable `start` is assigned to but never used
    |
315 |         try:
316 |             # Try to get replica status (which uses replicas for reads)
317 |             start = time.time()
    |             ^^^^^ F841
318 |             response = await self.client.get(f"{self.api_url}/health/replicas")
319 |             metric.end_time = time.time()
    |
    = help: Remove assignment to unused variable `start`

scripts/load_test_replicas.py:344:13: F841 Local variable `start` is assigned to but never used
    |
342 |             # This would execute a write operation (always goes to primary)
343 |             # For now, simulate with a health check endpoint
344 |             start = time.time()
    |             ^^^^^ F841
345 |             response = await self.client.post(
346 |                 f"{self.api_url}/health/check", json={"query": "write"}
    |
    = help: Remove assignment to unused variable `start`

server.py:334:1: E402 Module level import not at top of file
    |
334 | import sys
    | ^^^^^^^^^^ E402
    |

server.py:749:17: E722 Do not use bare `except`
    |
747 |                     json_data = json.loads(content.decode("utf-8"))
748 |                     print("DEBUG: Found valid JSON data")
749 |                 except:
    |                 ^^^^^^ E722
750 |                     print(f"DEBUG: Failed to parse JSON from {file.filename}")
751 |             elif file.filename.endswith(".zip"):
    |

server.py:894:9: E722 Do not use bare `except`
    |
892 |         try:
893 |             return list(csv.DictReader(io.StringIO(content)))
894 |         except:
    |         ^^^^^^ E722
895 |             return []
    |

tests/api_integration_tests/test_api_advanced_scenarios.py:273:16: E721 Use `is` and `is not` for type comparisons, or `isinstance()` for isinstance checks
    |
271 |         assert response2.status_code == 200
272 |         # Data structure should be consistent
273 |         assert type(response1.json()) == type(response2.json())
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ E721
    |

Found 56 errors (33 fixed, 23 remaining).
No fixes available (3 hidden fixes can be enabled with the `--unsafe-fixes` option).
- The E402 warnings (imports not at top of file) are intentional due to sys.path modification for vendor library

## QAQC Finding

Code quality issue - unused imports can cause confusion and hide real issues.

## Acceptance Criteria

- [ ] Run resume-api/api/async_pdf_routes.py:22:1: E402 Module level import not at top of file
   |
20 | sys.path.insert(0, str(lib_path))
21 |
22 | from lib.cli import ResumeGenerator
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
23 | from lib.queue import Job, JobState, JobPriority
24 | from config.dependencies import AuthorizedAPIKey, rate_limit
   |

resume-api/api/async_pdf_routes.py:23:1: E402 Module level import not at top of file
   |
22 | from lib.cli import ResumeGenerator
23 | from lib.queue import Job, JobState, JobPriority
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
24 | from config.dependencies import AuthorizedAPIKey, rate_limit
25 | from config import settings
   |

resume-api/api/async_pdf_routes.py:24:1: E402 Module level import not at top of file
   |
22 | from lib.cli import ResumeGenerator
23 | from lib.queue import Job, JobState, JobPriority
24 | from config.dependencies import AuthorizedAPIKey, rate_limit
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
25 | from config import settings
26 | from monitoring import logging_config
   |

resume-api/api/async_pdf_routes.py:25:1: E402 Module level import not at top of file
   |
23 | from lib.queue import Job, JobState, JobPriority
24 | from config.dependencies import AuthorizedAPIKey, rate_limit
25 | from config import settings
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
26 | from monitoring import logging_config
   |

resume-api/api/async_pdf_routes.py:26:1: E402 Module level import not at top of file
   |
24 | from config.dependencies import AuthorizedAPIKey, rate_limit
25 | from config import settings
26 | from monitoring import logging_config
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
27 |
28 | from .models import (
   |

resume-api/api/async_pdf_routes.py:28:1: E402 Module level import not at top of file
   |
26 |   from monitoring import logging_config
27 |
28 | / from .models import (
29 | |     SubmitPDFRenderJobRequest,
30 | |     SubmitPDFRenderJobResponse,
31 | |     JobStatusResponse,
32 | |     JobStatus,
33 | |     QueueStatsResponse,
34 | |     ErrorResponse,
35 | |     JobPriorityLevel,
36 | | )
   | |_^ E402
37 |
38 |   logger = logging_config.get_logger(__name__)
   |

resume-api/config/dependencies.py:100:16: F823 Local variable `verify_api_key` referenced before assignment
    |
 98 |     if settings.master_api_key:
 99 |         if is_hashed_key(settings.master_api_key):
100 |             if verify_api_key(x_api_key, settings.master_api_key):
    |                ^^^^^^^^^^^^^^ F823
101 |                 return x_api_key
102 |         else:
    |

resume-api/lib/utils/cache.py:588:1: E402 Module level import not at top of file
    |
588 | from functools import wraps
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
589 | from fastapi import Request, Response
    |

resume-api/lib/utils/cache.py:589:1: E402 Module level import not at top of file
    |
588 | from functools import wraps
589 | from fastapi import Request, Response
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
    |

resume-api/lib/utils/cache.py:589:21: F811 Redefinition of unused `Request` from line 18
    |
588 | from functools import wraps
589 | from fastapi import Request, Response
    |                     ^^^^^^^ F811
    |
    = help: Remove definition: `Request`

resume-api/lib/utils/cache.py:589:30: F811 Redefinition of unused `Response` from line 19
    |
588 | from functools import wraps
589 | from fastapi import Request, Response
    |                              ^^^^^^^^ F811
    |
    = help: Remove definition: `Response`

resume-api/tests/integration/test_api_key_management_e2e.py:324:1: E402 Module level import not at top of file
    |
323 | # Import User for test
324 | from database import User
    | ^^^^^^^^^^^^^^^^^^^^^^^^^ E402
    |

resume-api/tests/test_async_rendering.py:13:1: E402 Module level import not at top of file
   |
11 | sys.path.insert(0, str(resume_api_path))
12 |
13 | from main import app
   | ^^^^^^^^^^^^^^^^^^^^ E402
14 | from config.dependencies import get_api_key
   |

resume-api/tests/test_async_rendering.py:14:1: E402 Module level import not at top of file
   |
13 | from main import app
14 | from config.dependencies import get_api_key
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
15 |
16 | # Sample resume data for testing
   |

resume-api/tests/test_github_routes_v2.py:24:1: E402 Module level import not at top of file
   |
23 | # We need to mock the dependencies for testing
24 | from fastapi.security import HTTPBearer
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
25 |
26 | security = HTTPBearer()
   |

resume-cli/cli/utils/versioning.py:48:13: F841 Local variable `content` is assigned to but never used
   |
46 |         # Read current resume content
47 |         with open(self.yaml_path, "r", encoding="utf-8") as f:
48 |             content = f.read()
   |             ^^^^^^^ F841
49 |
50 |         # Get version number
   |
   = help: Remove assignment to unused variable `content`

resume_ai_lib/src/resume_ai_lib/ats_checker.py:460:13: F402 Import `field` from line 10 shadowed by loop variable
    |
459 |         basics = resume_data.get("basics", {})
460 |         for field in ["name", "summary", "headline"]:
    |             ^^^^^ F402
461 |             if basics.get(field):
462 |                 text_parts.append(str(basics[field]))
    |

scripts/load_test_replicas.py:317:13: F841 Local variable `start` is assigned to but never used
    |
315 |         try:
316 |             # Try to get replica status (which uses replicas for reads)
317 |             start = time.time()
    |             ^^^^^ F841
318 |             response = await self.client.get(f"{self.api_url}/health/replicas")
319 |             metric.end_time = time.time()
    |
    = help: Remove assignment to unused variable `start`

scripts/load_test_replicas.py:344:13: F841 Local variable `start` is assigned to but never used
    |
342 |             # This would execute a write operation (always goes to primary)
343 |             # For now, simulate with a health check endpoint
344 |             start = time.time()
    |             ^^^^^ F841
345 |             response = await self.client.post(
346 |                 f"{self.api_url}/health/check", json={"query": "write"}
    |
    = help: Remove assignment to unused variable `start`

server.py:334:1: E402 Module level import not at top of file
    |
334 | import sys
    | ^^^^^^^^^^ E402
    |

server.py:749:17: E722 Do not use bare `except`
    |
747 |                     json_data = json.loads(content.decode("utf-8"))
748 |                     print("DEBUG: Found valid JSON data")
749 |                 except:
    |                 ^^^^^^ E722
750 |                     print(f"DEBUG: Failed to parse JSON from {file.filename}")
751 |             elif file.filename.endswith(".zip"):
    |

server.py:894:9: E722 Do not use bare `except`
    |
892 |         try:
893 |             return list(csv.DictReader(io.StringIO(content)))
894 |         except:
    |         ^^^^^^ E722
895 |             return []
    |

tests/api_integration_tests/test_api_advanced_scenarios.py:273:16: E721 Use `is` and `is not` for type comparisons, or `isinstance()` for isinstance checks
    |
271 |         assert response2.status_code == 200
272 |         # Data structure should be consistent
273 |         assert type(response1.json()) == type(response2.json())
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ E721
    |

Found 23 errors.
No fixes available (3 hidden fixes can be enabled with the `--unsafe-fixes` option). to auto-fix unused imports
- [ ] Manually review any remaining warnings
- [ ] Run resume-api/api/async_pdf_routes.py:22:1: E402 Module level import not at top of file
   |
20 | sys.path.insert(0, str(lib_path))
21 |
22 | from lib.cli import ResumeGenerator
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
23 | from lib.queue import Job, JobState, JobPriority
24 | from config.dependencies import AuthorizedAPIKey, rate_limit
   |

resume-api/api/async_pdf_routes.py:23:1: E402 Module level import not at top of file
   |
22 | from lib.cli import ResumeGenerator
23 | from lib.queue import Job, JobState, JobPriority
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
24 | from config.dependencies import AuthorizedAPIKey, rate_limit
25 | from config import settings
   |

resume-api/api/async_pdf_routes.py:24:1: E402 Module level import not at top of file
   |
22 | from lib.cli import ResumeGenerator
23 | from lib.queue import Job, JobState, JobPriority
24 | from config.dependencies import AuthorizedAPIKey, rate_limit
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
25 | from config import settings
26 | from monitoring import logging_config
   |

resume-api/api/async_pdf_routes.py:25:1: E402 Module level import not at top of file
   |
23 | from lib.queue import Job, JobState, JobPriority
24 | from config.dependencies import AuthorizedAPIKey, rate_limit
25 | from config import settings
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
26 | from monitoring import logging_config
   |

resume-api/api/async_pdf_routes.py:26:1: E402 Module level import not at top of file
   |
24 | from config.dependencies import AuthorizedAPIKey, rate_limit
25 | from config import settings
26 | from monitoring import logging_config
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
27 |
28 | from .models import (
   |

resume-api/api/async_pdf_routes.py:28:1: E402 Module level import not at top of file
   |
26 |   from monitoring import logging_config
27 |
28 | / from .models import (
29 | |     SubmitPDFRenderJobRequest,
30 | |     SubmitPDFRenderJobResponse,
31 | |     JobStatusResponse,
32 | |     JobStatus,
33 | |     QueueStatsResponse,
34 | |     ErrorResponse,
35 | |     JobPriorityLevel,
36 | | )
   | |_^ E402
37 |
38 |   logger = logging_config.get_logger(__name__)
   |

resume-api/config/dependencies.py:100:16: F823 Local variable `verify_api_key` referenced before assignment
    |
 98 |     if settings.master_api_key:
 99 |         if is_hashed_key(settings.master_api_key):
100 |             if verify_api_key(x_api_key, settings.master_api_key):
    |                ^^^^^^^^^^^^^^ F823
101 |                 return x_api_key
102 |         else:
    |

resume-api/lib/utils/cache.py:588:1: E402 Module level import not at top of file
    |
588 | from functools import wraps
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
589 | from fastapi import Request, Response
    |

resume-api/lib/utils/cache.py:589:1: E402 Module level import not at top of file
    |
588 | from functools import wraps
589 | from fastapi import Request, Response
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
    |

resume-api/lib/utils/cache.py:589:21: F811 Redefinition of unused `Request` from line 18
    |
588 | from functools import wraps
589 | from fastapi import Request, Response
    |                     ^^^^^^^ F811
    |
    = help: Remove definition: `Request`

resume-api/lib/utils/cache.py:589:30: F811 Redefinition of unused `Response` from line 19
    |
588 | from functools import wraps
589 | from fastapi import Request, Response
    |                              ^^^^^^^^ F811
    |
    = help: Remove definition: `Response`

resume-api/tests/integration/test_api_key_management_e2e.py:324:1: E402 Module level import not at top of file
    |
323 | # Import User for test
324 | from database import User
    | ^^^^^^^^^^^^^^^^^^^^^^^^^ E402
    |

resume-api/tests/test_async_rendering.py:13:1: E402 Module level import not at top of file
   |
11 | sys.path.insert(0, str(resume_api_path))
12 |
13 | from main import app
   | ^^^^^^^^^^^^^^^^^^^^ E402
14 | from config.dependencies import get_api_key
   |

resume-api/tests/test_async_rendering.py:14:1: E402 Module level import not at top of file
   |
13 | from main import app
14 | from config.dependencies import get_api_key
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
15 |
16 | # Sample resume data for testing
   |

resume-api/tests/test_github_routes_v2.py:24:1: E402 Module level import not at top of file
   |
23 | # We need to mock the dependencies for testing
24 | from fastapi.security import HTTPBearer
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
25 |
26 | security = HTTPBearer()
   |

resume-cli/cli/utils/versioning.py:48:13: F841 Local variable `content` is assigned to but never used
   |
46 |         # Read current resume content
47 |         with open(self.yaml_path, "r", encoding="utf-8") as f:
48 |             content = f.read()
   |             ^^^^^^^ F841
49 |
50 |         # Get version number
   |
   = help: Remove assignment to unused variable `content`

resume_ai_lib/src/resume_ai_lib/ats_checker.py:460:13: F402 Import `field` from line 10 shadowed by loop variable
    |
459 |         basics = resume_data.get("basics", {})
460 |         for field in ["name", "summary", "headline"]:
    |             ^^^^^ F402
461 |             if basics.get(field):
462 |                 text_parts.append(str(basics[field]))
    |

scripts/load_test_replicas.py:317:13: F841 Local variable `start` is assigned to but never used
    |
315 |         try:
316 |             # Try to get replica status (which uses replicas for reads)
317 |             start = time.time()
    |             ^^^^^ F841
318 |             response = await self.client.get(f"{self.api_url}/health/replicas")
319 |             metric.end_time = time.time()
    |
    = help: Remove assignment to unused variable `start`

scripts/load_test_replicas.py:344:13: F841 Local variable `start` is assigned to but never used
    |
342 |             # This would execute a write operation (always goes to primary)
343 |             # For now, simulate with a health check endpoint
344 |             start = time.time()
    |             ^^^^^ F841
345 |             response = await self.client.post(
346 |                 f"{self.api_url}/health/check", json={"query": "write"}
    |
    = help: Remove assignment to unused variable `start`

server.py:334:1: E402 Module level import not at top of file
    |
334 | import sys
    | ^^^^^^^^^^ E402
    |

server.py:749:17: E722 Do not use bare `except`
    |
747 |                     json_data = json.loads(content.decode("utf-8"))
748 |                     print("DEBUG: Found valid JSON data")
749 |                 except:
    |                 ^^^^^^ E722
750 |                     print(f"DEBUG: Failed to parse JSON from {file.filename}")
751 |             elif file.filename.endswith(".zip"):
    |

server.py:894:9: E722 Do not use bare `except`
    |
892 |         try:
893 |             return list(csv.DictReader(io.StringIO(content)))
894 |         except:
    |         ^^^^^^ E722
895 |             return []
    |

tests/api_integration_tests/test_api_advanced_scenarios.py:273:16: E721 Use `is` and `is not` for type comparisons, or `isinstance()` for isinstance checks
    |
271 |         assert response2.status_code == 200
272 |         # Data structure should be consistent
273 |         assert type(response1.json()) == type(response2.json())
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ E721
    |

Found 23 errors.
No fixes available (3 hidden fixes can be enabled with the `--unsafe-fixes` option). - target 0 F401 warnings
- [ ] Keep E402 warnings (they're intentional)

## Fix Command

https://github.com/anchapin/ResumeAI/issues/730
api/async_pdf_routes.py:22:1: E402 Module level import not at top of file
   |
20 | sys.path.insert(0, str(lib_path))
21 |
22 | from lib.cli import ResumeGenerator
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
23 | from lib.queue import Job, JobState, JobPriority
24 | from config.dependencies import AuthorizedAPIKey, rate_limit
   |

api/async_pdf_routes.py:23:1: E402 Module level import not at top of file
   |
22 | from lib.cli import ResumeGenerator
23 | from lib.queue import Job, JobState, JobPriority
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
24 | from config.dependencies import AuthorizedAPIKey, rate_limit
25 | from config import settings
   |

api/async_pdf_routes.py:24:1: E402 Module level import not at top of file
   |
22 | from lib.cli import ResumeGenerator
23 | from lib.queue import Job, JobState, JobPriority
24 | from config.dependencies import AuthorizedAPIKey, rate_limit
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
25 | from config import settings
26 | from monitoring import logging_config
   |

api/async_pdf_routes.py:25:1: E402 Module level import not at top of file
   |
23 | from lib.queue import Job, JobState, JobPriority
24 | from config.dependencies import AuthorizedAPIKey, rate_limit
25 | from config import settings
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
26 | from monitoring import logging_config
   |

api/async_pdf_routes.py:26:1: E402 Module level import not at top of file
   |
24 | from config.dependencies import AuthorizedAPIKey, rate_limit
25 | from config import settings
26 | from monitoring import logging_config
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
27 |
28 | from .models import (
   |

api/async_pdf_routes.py:28:1: E402 Module level import not at top of file
   |
26 |   from monitoring import logging_config
27 |
28 | / from .models import (
29 | |     SubmitPDFRenderJobRequest,
30 | |     SubmitPDFRenderJobResponse,
31 | |     JobStatusResponse,
32 | |     JobStatus,
33 | |     QueueStatsResponse,
34 | |     ErrorResponse,
35 | |     JobPriorityLevel,
36 | | )
   | |_^ E402
37 |
38 |   logger = logging_config.get_logger(__name__)
   |

config/dependencies.py:100:16: F823 Local variable `verify_api_key` referenced before assignment
    |
 98 |     if settings.master_api_key:
 99 |         if is_hashed_key(settings.master_api_key):
100 |             if verify_api_key(x_api_key, settings.master_api_key):
    |                ^^^^^^^^^^^^^^ F823
101 |                 return x_api_key
102 |         else:
    |

lib/utils/cache.py:588:1: E402 Module level import not at top of file
    |
588 | from functools import wraps
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
589 | from fastapi import Request, Response
    |

lib/utils/cache.py:589:1: E402 Module level import not at top of file
    |
588 | from functools import wraps
589 | from fastapi import Request, Response
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
    |

lib/utils/cache.py:589:21: F811 Redefinition of unused `Request` from line 18
    |
588 | from functools import wraps
589 | from fastapi import Request, Response
    |                     ^^^^^^^ F811
    |
    = help: Remove definition: `Request`

lib/utils/cache.py:589:30: F811 Redefinition of unused `Response` from line 19
    |
588 | from functools import wraps
589 | from fastapi import Request, Response
    |                              ^^^^^^^^ F811
    |
    = help: Remove definition: `Response`

tests/integration/test_api_key_management_e2e.py:324:1: E402 Module level import not at top of file
    |
323 | # Import User for test
324 | from database import User
    | ^^^^^^^^^^^^^^^^^^^^^^^^^ E402
    |

tests/test_async_rendering.py:13:1: E402 Module level import not at top of file
   |
11 | sys.path.insert(0, str(resume_api_path))
12 |
13 | from main import app
   | ^^^^^^^^^^^^^^^^^^^^ E402
14 | from config.dependencies import get_api_key
   |

tests/test_async_rendering.py:14:1: E402 Module level import not at top of file
   |
13 | from main import app
14 | from config.dependencies import get_api_key
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
15 |
16 | # Sample resume data for testing
   |

tests/test_github_routes_v2.py:24:1: E402 Module level import not at top of file
   |
23 | # We need to mock the dependencies for testing
24 | from fastapi.security import HTTPBearer
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
25 |
26 | security = HTTPBearer()
   |

Found 15 errors.

## Files

- All Python files in

--- a/resume-cli/cli/utils/versioning.py
+++ b/resume-cli/cli/utils/versioning.py
@@ -43,9 +43,9 @@ class ResumeVersionManager:
         Returns:
             Dictionary with version info
         """
-        # Read current resume content
+        # Read current resume content to ensure it's accessible
         with open(self.yaml_path, "r", encoding="utf-8") as f:
-            content = f.read()
+            f.read()
 
         # Get version number
         version_number = self._get_next_version_number()

--- a/resume_ai_lib/src/resume_ai_lib/ats_checker.py
+++ b/resume_ai_lib/src/resume_ai_lib/ats_checker.py
@@ -457,13 +457,13 @@ class ATSCompatibilityChecker:
         text_parts = []
 
         basics = resume_data.get("basics", {})
-        for field in ["name", "summary", "headline"]:
-            if basics.get(field):
-                text_parts.append(str(basics[field]))
+        for field_name in ["name", "summary", "headline"]:
+            if basics.get(field_name):
+                text_parts.append(str(basics[field_name]))
 
-        for field in ["work", "experience"]:
-            if field in resume_data and isinstance(resume_data[field], list):
-                for exp in resume_data[field]:
+        for section in ["work", "experience"]:
+            if section in resume_data and isinstance(resume_data[section], list):
+                for exp in resume_data[section]:
                     if isinstance(exp, dict):
                         for key in [
                             "position",

--- a/scripts/load_test_replicas.py
+++ b/scripts/load_test_replicas.py
@@ -314,7 +314,6 @@ class ReplicaLoadTester:
 
         try:
             # Try to get replica status (which uses replicas for reads)
-            start = time.time()
             response = await self.client.get(f"{self.api_url}/health/replicas")
             metric.end_time = time.time()
 
@@ -341,7 +340,6 @@ class ReplicaLoadTester:
         try:
             # This would execute a write operation (always goes to primary)
             # For now, simulate with a health check endpoint
-            start = time.time()
             response = await self.client.post(
                 f"{self.api_url}/health/check", json={"query": "write"}
             )

--- a/server.py
+++ b/server.py
@@ -746,7 +746,7 @@ async def import_linkedin_file(files: List[UploadFile] = File(...)):
                 try:
                     json_data = json.loads(content.decode("utf-8"))
                     print("DEBUG: Found valid JSON data")
-                except:
+                except json.JSONDecodeError:
                     print(f"DEBUG: Failed to parse JSON from {file.filename}")
             elif file.filename.endswith(".zip"):
                 print("DEBUG: Processing ZIP file")
@@ -891,7 +891,7 @@ def _process_linkedin_csvs(csv_files):
 
         try:
             return list(csv.DictReader(io.StringIO(content)))
-        except:
+        except csv.Error:
             return []
 
     # Map filenames to keys

--- a/tests/api_integration_tests/test_api_advanced_scenarios.py
+++ b/tests/api_integration_tests/test_api_advanced_scenarios.py
@@ -271,7 +271,7 @@ class TestDataIntegrity:
         assert response1.status_code == 200
         assert response2.status_code == 200
         # Data structure should be consistent
-        assert type(response1.json()) == type(response2.json())
+        assert type(response1.json()) is type(response2.json())
 
 
 class TestResponseConsistency:

